File: csharp/src/code-smells/dispensables/LazyClass.cs
namespace RefactoringAvanzado.CodeSmells.Dispensables;

public record Address(string Name, string Line1, string? City);

public class ShippingLabelBuilder
{
    public string Build(Address a)
    {
        return $"{a.Name} — {a.Line1}{(a.City != null ? $", {a.City}" : "")}";
    }
}

public static class LazyClass
{
    public static void PrintShippingLabel()
    {
        var address = new Address("John Doe", "123 Main St", "New York");

        var labelBuilder = new ShippingLabelBuilder();
        var label = labelBuilder.Build(address);
        Console.WriteLine(label);
    }
}

File: typescript/src/code-smells/dispensables/lazy-class.ts
export type Address = { name: string; line1: string; city?: string }

export class ShippingLabelBuilder {
  build(a: Address): string {
    return `${a.name} — ${a.line1}${a.city ? ', ' + a.city : ''}`
  }
}

export function printShippingLabel() {
  const address: Address = {
    name: 'John Doe',
    line1: '123 Main St',
    city: 'New York',
  }

  const labelBuilder = new ShippingLabelBuilder()
  const label = labelBuilder.build(address)
  console.log(label)
}
